CMakeLists.txt

```diff
@@ -79,6 +79,47 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
   add_compile_options(-Wno-psabi)
 endif()
 
+execute_process(COMMAND git rev-parse --abbrev-ref HEAD
+    WORKING_DIRECTORY ${QuokkaCode_SOURCE_DIR}
+    OUTPUT_VARIABLE QUOKKA_GIT_BRANCH
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+execute_process(COMMAND git rev-parse HEAD
+    WORKING_DIRECTORY ${QuokkaCode_SOURCE_DIR}
+    OUTPUT_VARIABLE QUOKKA_GIT_HASH
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+execute_process(COMMAND git rev-parse HEAD
+    WORKING_DIRECTORY ${QuokkaCode_SOURCE_DIR}/extern/amrex
+    OUTPUT_VARIABLE AMREX_GIT_HASH
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+# Check if the working directory is dirty
+execute_process(
+    COMMAND git status --porcelain -uno
+    WORKING_DIRECTORY ${QuokkaCode_SOURCE_DIR}
+    OUTPUT_VARIABLE GIT_STATUS_OUTPUT
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+# Set a flag if the working directory is dirty
+if(NOT GIT_STATUS_OUTPUT STREQUAL "")
+    set(GIT_DIRTY 1)
+    message(WARNING "This Quokka repository has untracked changes not reflected in its git history! That means that the exact provenance of plotfiles and checkpoints CANNOT be inferred from their metadata.\nIt is STRONGLY recommended that you commit your changes to a branch locally, then re-run CMake and recompile Quokka in order to obtain reproducible simulation outputs.")
```

Comment:
I've updated this text to make it clearer that changes need only be committed locally to any branch.